contracts/core/Registry.sol

```diff
@@ -138,10 +140,7 @@ contract Registry is IRegistry, Native, AccessControl, Transfer {
         Identity storage identity = identitiesById[_identityId];
         identity.name = _name;
 
-        // clear old anchor
-        anchorToIdentityId[identity.anchor] = bytes32(0);
-
-        // set new anchor
+        // set new anchor while preserving old anchor
         identity.anchor = anchor;
         anchorToIdentityId[anchor] = _identityId;
```

Comment:
Hmm... this is tricky. Let's think through it...
- anchorToIdentityId is saying that a given anchor corresponds with an identity
- once an identity's info changes (ie name change), then the anchor changes too
- this is because we might have attestations saying, for example, "we've verified this anchor has a valid name, because they've shown us their twitter"
- so if the old anchor can point to the new identity, it might keep that attestation, which is a problem

So we can't keep the anchorToIdentityId[identity.anchor] actively pointing to the new identity.

Fortunately, I don't think we have to. If we were looking up the owner of an anchor as (psuedocode) `isOwnerofIdenity(anchorToIdentityId[msg.sender])` then we'd need to solve this. But since the anchor is saving the `identityId` locally, I believe the old code would solve itself.

- Anchor address saves `identityId`
- If a name change happens, the `anchorToIdentityId` is deleted
- But if someone calls execute on it, as long as they're the owner of the `identityId` (which hasn't changed), then it'll still pass
- So everything should be fine

Do you agree?

Reply:
Ah you are right here ! Preserving the old mapping will cause issues.

Reply:
fixed via 1006b38